### PR TITLE
Docker workflow

### PR DESCRIPTION
**Pull request by johnkgamble**
Example of how to use Python in a Docker container.

### CODE REVIEW

File: deployment/docker_python.sh [hunk 1]
```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+# ADAPT THE CONFIGURATION VARIABLES BELOW TO YOUR USE CASE! You will at least
```
Comment (likewei92) [inline deployment/docker_python.sh]: why can't this script be run in place?

Comment (belabauer) [reply deployment/docker_python.sh]: What do you mean?

Comment (likewei92) [reply deployment/docker_python.sh]: Never mind I was just confused about why the DOCKER_PYTHON_SCRIPT needs to be specified

Comment (belabauer) [reply deployment/docker_python.sh]: I guess one could pass the part that is run inside of Docker as a single string to the docker run command, but I thought that would actually make for a less readable script without any obvious advantages.

**Comment by likewei92** [comment]
I think the more "elegant" way to do it is through a docker entrypoint script, with the only downside being the user of the docker container needs to know to open the 9000 port for jupyter (if needed)

**Comment by belabauer** [comment]
I'm not sure how to achieve the same with an entrypoint script: first of all, this carries some information (like the numerical user ID) from the invoking environment into docker to make sure that matches (otherwise the user ends up with a bunch of files owned either by root or some random other user in their home). Also, note that this doesn't work just for opening jupyter notebooks but allows much more general uses.

**Comment by likewei92** [comment]
maybe I'm just missing some context here but why do we need to use the same userId/userName within the docker container?

**Comment by belabauer** [comment]
So that the user can use the files in their home directory transparently.

**Comment by likewei92** [comment]
Oh ic. In that case I think a slightly better code organization is to move the `if [ -f /.dockerenv ]` part of the script into an entrypoint script (rather than relying on the `.dockerenv` file to tell if we're inside the container or not, which is what it seems to be doing?) and then keeping the rest of the script as is. I think that way the script is easier to understand and we have a clearer separation of concerns between what's outside and what's inside the container

However if we're not too particular I also don't mind just checking this in :P

**Comment by likewei92** [comment]
I guess more specifically, this script is being called twice (once upon invocation outside the container, once inside the container to run python) to do different things (set up env variables to run the docker image vs running python within the image), which seems a little hacky to me. I think it would be cleaner/easier for someone else to come take a look and understand what's going on if we split those two things up

**Pull request merged**